### PR TITLE
CO-2519: Add buttons to print all medical/feedback surveys

### PR DESCRIPTION
**Pull request by Nooka10**

### CODE REVIEW

**Comment by ecino** [comment]
Some flake8/formatting issues:
```
website_event_compassion/models/event_event.py:92:27: E131 continuation line unaligned for hanging indent
website_event_compassion/models/event_event.py:93:35: E126 continuation line over-indented for hanging indent
website_event_compassion/models/event_event.py:93:89: E501 line too long (91 > 88 characters)
website_event_compassion/models/event_event.py:100:39: E127 continuation line over-indented for visual indent
website_event_compassion/models/event_event.py:100:89: E501 line too long (92 > 88 characters)
website_event_compassion/models/event_event.py:106:41: E127 continuation line over-indented for visual indent
website_event_compassion/models/event_event.py:106:89: E501 line too long (97 > 88 characters)
website_event_compassion/models/event_event.py:117:39: E127 continuation line over-indented for visual indent
website_event_compassion/models/event_event.py:117:89: E501 line too long (96 > 88 characters)
```

File: website_event_compassion/views/event_event_view.xml [hunk 1]
```diff
@@ -8,6 +8,8 @@
             <button name="button_confirm" position="before">
                 <button string="Create Information Party" name="button_info_party" type="object" class="oe_highlight" groups="base.group_user" attrs="{'invisible': ['|', ('event_type_id', '!=', %(website_event_compassion.event_type_group_visit)d), ('state', '!=', 'confirm')]}"/>
                 <button string="After trip Party" name="button_info_party" type="object" class="oe_highlight" groups="base.group_user" attrs="{'invisible': ['|', ('event_type_id', '!=', %(website_event_compassion.event_type_group_visit)d), ('state', '!=', 'done')]}"/>
+                <button string="Print Medical Surveys" name="button_print_medical_surveys" type="object" class="oe_highlight" groups="base.group_user"/>
```
Comment (ecino) [inline website_event_compassion/views/event_event_view.xml:11]: You added the button but the method is not defined anywhere! Did you forget to commit some files?

**Review by ecino — CHANGES_REQUESTED** [review]

File: website_event_compassion/views/event_event_view.xml (hunk 1 above)
Comment (Nooka10) [reply website_event_compassion/views/event_event_view.xml:11]: Oups. I erased them by error in the previous commit... -_-"

Corrected in commit e5274e10cc3fa4935432968cf4ae9d86020857dc

File: website_event_compassion/models/event_event.py [hunk 2]
```diff
@@ -158,6 +157,28 @@ def button_info_party(self):
             'target': 'new',
         }
 
+    @api.multi
+    def button_print_medical_surveys(self):
+        medical_survey_ids = self.compassion_event_id.registration_ids \
```
Comment (ecino) [inline website_event_compassion/models/event_event.py]:
```suggestion
        medical_survey_ids = self.registration_ids \
```

File: website_event_compassion/models/event_event.py [hunk 3]
```diff
@@ -158,6 +157,28 @@ def button_info_party(self):
...
+            .mapped('medical_survey_id') \
+            .filtered(lambda p: p.state and p.state == 'done')
```
Comment (ecino) [inline website_event_compassion/models/event_event.py]:
```suggestion
            .filtered(lambda p: p.state == 'done')
```

File: website_event_compassion/models/event_event.py [hunk 4]
```diff
@@ -158,6 +157,28 @@ def button_info_party(self):
...
+        if medical_survey_ids:
+            return self.env['report'].get_action(
+                medical_survey_ids, 'survey_phone.survey_user_input')
+        else:
+            raise UserWarning('There is no medical survey to print')
```
Comment (ecino) [inline website_event_compassion/models/event_event.py]:
```
from odoo.exceptions import UserError
from odoo import _
```
```suggestion
            raise UserError(_('There is no medical survey to print'))
```

File: website_event_compassion/models/event_event.py [hunk 5]
```diff
@@ -158,6 +157,28 @@ def button_info_party(self):
...
+
+    @api.multi
+    def button_print_feedback_surveys(self):
+        feedback_survey_ids = self.compassion_event_id.registration_ids \
```
Comment (ecino) [inline website_event_compassion/models/event_event.py]:
```suggestion
        feedback_survey_ids = self.registration_ids \
```

File: website_event_compassion/models/event_event.py [hunk 6]
```diff
@@ -158,6 +157,28 @@ def button_info_party(self):
...
+            .mapped('feedback_survey_id') \
+            .filtered(lambda p: p.state and p.state == 'done')
```
Comment (ecino) [inline website_event_compassion/models/event_event.py]:
```suggestion
            .filtered(lambda p: p.state == 'done')
```

File: website_event_compassion/models/event_event.py [hunk 7]
```diff
@@ -158,6 +157,28 @@ def button_info_party(self):
...
+        if feedback_survey_ids:
+            return self.env['report'].get_action(
+                feedback_survey_ids, 'survey_phone.survey_user_input')
+        else:
+            raise UserWarning('There is no feedback survey to print')
```
Comment (ecino) [inline website_event_compassion/models/event_event.py]:
```suggestion
            raise UserError(_('There is no feedback survey to print'))
```

**Review by ecino — CHANGES_REQUESTED** [review]
Thanks! Here are some suggestions.

**Pull request closed without merging**